src/Controller/GoogleController.php

```diff
@@ -0,0 +1,45 @@
+<?php
+namespace App\Controller;
+
+    use KnpU\OAuth2ClientBundle\Client\ClientRegistry;
+    use League\OAuth2\Client\Provider\Exception\IdentityProviderException;
+    use Symfony\Bundle\FrameworkBundle\Controller\AbstractController;
+    use Symfony\Component\HttpFoundation\Request;
+    use Symfony\Component\Routing\Annotation\Route;
+
+class GoogleController extends AbstractController
+{
+    /**
+     * Link to this controller to start the "connect" process
+     * @param ClientRegistry $clientRegistry
+     *
+     * @Route("/connect/google", name="connect_google_start")
+     *
+     * @return \Symfony\Component\HttpFoundation\RedirectResponse
+     */
+    public function connectAction(ClientRegistry $clientRegistry)
+    {
+        return $clientRegistry
+            ->getClient('google')
+            ->redirect([
+                'profile', 'email' // the scopes you want to access
+            ])
+            ;
+    }
+
+    /**
+     * After going to Google, you're redirected back here
+     * because this is the "redirect_route" you configured
+     * in config/packages/knpu_oauth2_client.yaml
+     *
+     * @param Request $request
+     * @param ClientRegistry $clientRegistry
+     *
+     * @Route("/connect/google/check", name="connect_google_check")
+     * @return \Symfony\Component\HttpFoundation\RedirectResponse
+     */
+    public function connectCheckAction(Request $request, ClientRegistry $clientRegistry)
+    {
+        return $this->redirectToRoute('home');
+    }
+}
```

Comment:
Currently this does full page redirects, which will be not very good for the user from UX point of view, would it be possible to open this in blank tab and after confirmation just return 200, that user still would be in state with open modal ready to write comment?